examples/chart/teleport-cluster/templates/auth/_config.common.tpl

```diff
@@ -0,0 +1,64 @@
+{{- define "teleport-cluster.auth.config.common" -}}
+{{- $authentication := mustMergeOverwrite .Values.authentication (default dict .Values.authenticationSecondFactor) -}}
+{{- $logLevel := (coalesce .Values.logLevel .Values.log.level "INFO") -}}
+version: v3
+teleport:
+  log:
+    severity: {{ $logLevel }}
+    output: {{ .Values.log.output }}
+    format:
+      output: {{ .Values.log.format }}
+      extra_fields: {{ .Values.log.extraFields | toJson }}
+kubernetes_service:
+  enabled: true
+  listen_addr: 0.0.0.0:3026
+  public_addr: "{{ .Release.Name }}-auth.{{ .Release.Namespace }}.svc.cluster.local:3026"
```

Comment:
I am not happy with the hardcoded cluster domain `cluster.local`. Should this be configurable in value ? Do we want to rely on the ndots behaviour? I'm not sure about how many requests and how many resolutions the proxy performs but this tends to scale terribly.

Reply:
I have an idea that will not involve exposing a value but will allow users to override it: I'll put this in a helper template (it's used at 2 or 3 places, so it kinda makes sense to compute it once).

By the way, if you squint hard enough, the chart not only supports `standalone`, `aws`, `gcp` and `scratch` modes, but lets users build their own modes with templating.

If you create your own chart, declare ours as a dependency, and define `teleport-cluster.auth.config.mymode` (or redefine `teleport-cluster.auth.config.common`), you can totally set `chartMode: mymode`. I really don't want to document this because this is not needed by 99% of the users and they will abuse it, but this is definitely a nice trick for Helm power users.

Reply:
If I understand correctly the init-container would mount the config, resolve `{{.Release.Name}}-auth`, sed the original config to an `empty-dir` volume and replace a kubernetes_service.public_addr placeholder.

Reply:
Yes, that's my idea

Reply:
I think we'll need to build a FQDN it anyway for the proxy initContainer checking if the auth server are updated.